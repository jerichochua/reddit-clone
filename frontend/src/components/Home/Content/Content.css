.content {
  flex: 1;
}

.content ul {
  list-style: none;
  border: 1px solid rgb(235, 237, 240);
  border-radius: 2px;
}

.content li:not(:first-child) {
  border-top: 1px solid rgb(235, 237, 240);
}

.post-container {
  display: flex;
}

.post-item {
  display: flex;
  flex: 1;
  flex-direction: column;
  border-left: 1px solid rgb(235, 237, 240);
  padding: 8px;
}

.votes {
  display: flex;
  flex-direction: column;
  padding: 4px;
  font-size: 12px;
  font-weight: 500;
  color: rgb(69, 79, 91);
  align-items: center;
  line-height: 25px;
  width: 30px;
}

.upvote-btn {
  height: 22px;
  width: 22px;
  border: 0px none;
  background-color: transparent;
}

.upvote-btn::after {
  border: solid #b6b6b6;
  border-width: 2px 2px 0 0;
  width: 8px;
  height: 8px;
  display: block;
  position: relative;
  left: 6px;
  top: 3px;
  content: '';
  transform: rotate(-45deg);
}

.downvote-btn {
  height: 22px;
  width: 22px;
  border: 0px none;
  background-color: transparent;
}

.downvote-btn::after {
  border: solid #b6b6b6;
  border-width: 0 0 2px 2px;
  width: 8px;
  height: 8px;
  display: block;
  position: relative;
  left: 6px;
  top: -2px;
  content: '';
  transform: rotate(-45deg);
}

.title {
  font-weight: 500;
  flex-grow: 1;
}

.caption {
  font-size: 13px;
  color: rgb(129, 142, 153);
  flex-grow: 1;
}

.links {
  font-size: 13px;
  flex-grow: 1;
}
