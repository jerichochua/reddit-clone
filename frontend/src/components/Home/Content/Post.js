import React from 'react';

export default function Post(props) {
  return (
    <li>
      <div className='post-container'>
        <div className='votes'>
          <button className='upvote-btn'></button>
          <span>{props.votes}</span>
          <button className='downvote-btn'></button>
        </div>
        <div className='post-item'>
          <p className='title'>{props.title}</p>
          <p className='caption'>
            by {props.author} {props.timestamp}
          </p>
          <p className='links'>{props.comments} comments</p>
        </div>
      </div>
    </li>
  );
}
