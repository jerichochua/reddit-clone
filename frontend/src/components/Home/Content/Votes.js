import React from 'react';

export default function Votes(props) {
  return (
    <div className='votes'>
      <button className='upvote-btn'></button>
      <span>{props.votes}</span>
      <button className='downvote-btn'></button>
    </div>
  );
}
