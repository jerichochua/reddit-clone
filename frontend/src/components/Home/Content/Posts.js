import React from 'react';
import { Link } from 'react-router-dom';
import Votes from '../../Vote/Votes';

const Posts = ({
  postId,
  title,
  author,
  score,
  timestamp,
  comments,
  type,
  url,
}) => {
  return (
    <li>
      <div className='post-container'>
        <Votes score={score} />
        <div className='post-item'>
          <div className='title'>
            {type === 'text' && <Link to={`/posts/${postId}`}>{title}</Link>}
            {type === 'link' && (
              <a href={url} target='_blank' rel='noreferrer'>
                {title}
              </a>
            )}
          </div>
          {type === 'link' && <div className='post-url'>{url}</div>}
          <div className='caption'>
            <span>by </span>
            <Link to={`/user/${author}`}>{author}</Link>
            <span> {timestamp}</span>
          </div>
          <div>
            <Link className='links' to={`/posts/${postId}`}>
              {comments} comment{comments === 1 ? '' : 's'}
            </Link>
          </div>
        </div>
      </div>
    </li>
  );
};

export default Posts;
