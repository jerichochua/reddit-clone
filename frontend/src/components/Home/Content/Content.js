import React, { useEffect, useState } from 'react';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import { get } from '../../../util/api';
import Posts from './Posts';
import './Content.css';

dayjs.extend(relativeTime);

const Content = (props) => {
  const [posts, setPosts] = useState([]);

  useEffect(() => {
    const fetchPosts = async () => {
      const response = await get('posts');
      setPosts(response);
    };
    fetchPosts();
  }, []);

  const postsList = posts.map((post) => {
    return (
      <Posts
        key={post.id}
        postid={post.id}
        title={post.title}
        author={post.author}
        timestamp={dayjs(post.created_at).fromNow()}
        votes={post.score}
        comments={parseInt(post.comments)}
      />
    );
  });

  return (
    <div className='content'>
      {posts.length === 0 ? <p>No posts to show</p> : <ul>{postsList}</ul>}
    </div>
  );
};

export default Content;
