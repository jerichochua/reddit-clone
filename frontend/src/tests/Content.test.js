import renderer from 'react-test-renderer';
import Content from '../components/Home/Content/Content';

it('renders correctly', () => {
  const tree = renderer.create(<Content />).toJSON();
  expect(tree).toMatchSnapshot();
});
