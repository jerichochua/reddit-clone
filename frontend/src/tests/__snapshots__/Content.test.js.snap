// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="empty-content"
  >
    <p>
      No content found. Please try a different page.
    </p>
    <p
      className="empty-message"
    >
      No posts to show
    </p>
  </div>
</div>
`;
